PerformanceTest_Static_Pressure_PlantarDorsal_ShinCalf_v2.py

```diff
@@ -0,0 +1,413 @@
+
+# Read in files
+# only read .asc files for this work
+
+fPath = 'C:\\Users\\bethany.kilpatrick\\BOA Technology Inc\\PFL Team - General\\Testing Segments\\Snow Performance\\2025_Performance_LassoPressure_Ride\\XSENSOR\\Exported\\'
+fileExt = r".csv"
+
+entries = [fName for fName in os.listdir(fPath) if fName.endswith(fileExt) and '0_' not in fName]
+entries_FrontBack = [fName for fName in os.listdir(fPath) if (fName.endswith(fileExt) and 'FrontBack' in fName)]
+entries_FootDorsum = [fName for fName in os.listdir(fPath) if (fName.endswith(fileExt) and 'FootDorsum' in fName)  ]
+
+
+### set plot font size ###
+SMALL_SIZE = 14
+MEDIUM_SIZE = 16
+BIGGER_SIZE = 18
+
+plt.rc('font', size=SMALL_SIZE)          # controls default text sizes
+plt.rc('axes', titlesize=SMALL_SIZE)     # fontsize of the axes title
+plt.rc('axes', labelsize=MEDIUM_SIZE)    # fontsize of the x and y labels
+plt.rc('xtick', labelsize=SMALL_SIZE)    # fontsize of the tick labels
+plt.rc('ytick', labelsize=SMALL_SIZE)    # fontsize of the tick labels
+plt.rc('legend', fontsize=SMALL_SIZE)    # legend fontsize
+plt.rc('figure', titlesize=BIGGER_SIZE)  # fontsize of the figure title
+
+@dataclass
+class FB_avgData:
+    
+    avgshinMat: np.array 
+
+    
+    avgcalfMat: np.array   
+
+    
+    config: str
+    subject: str
+    
+    # dat_FD: pd.DataFrame
+    dat_FB: pd.DataFrame
+    #entire stored dataframe. 
+   
+    # def plotAvgFBPressure(self):
+    #     fig, (ax1, ax2) = plt.subplots(1,2)
+    #     ax1 = sns.heatmap(self.avgshinMat, ax = ax1, cmap="mako", vmin = 0, vmax = np.max(self.avgshinMat) * 2)
+    #     ax1.set(xticklabels=[])
+    #     ax1.set_title('Shin Pressure') 
+    #     ax2 = sns.heatmap(self.avgcalfMat, ax = ax2, cmap="mako", vmin = 0, vmax = np.max(self.avgcalfMat) * 2)
+    #     ax2.set(xticklabels=[])
+    #     ax2.set_title('Calf Pressure') 
+    #     plt.suptitle(self.config)
+    #     plt.tight_layout() 
+        
+    #     saveFolder= fPath + '2DPlots'
+        
+    #     if os.path.exists(saveFolder) == False:
+    #         os.mkdir(saveFolder)
+            
+    #     plt.savefig(saveFolder + '/' + self.subject + self.config + '.png')
+         
+    #     return fig  
+    
+def createAvg_FB_Mat(inputName):
+    """ 
+    Reads in file, creates average matrix data, in the shape of the pressure sensor(s), to be plotted and features
+    are extracted. The result is a dataclass which can be used for further plotting
+    
+    inputName: string
+        filename of data static trial you are processing. 
+    """
+   
+    # entries_frontBack = [fName for fName in os.listdir(fPath) if (fName.endswith(fileExt) and 'FrontBack' in fName) ]
+    # entries_FootDorsum = [fName for fName in os.listdir(fPath) if (fName.endswith(fileExt) and 'FootDorsum' in fName) ] 
+
+    
+
+    # Front back 
+   
+
+    subj = inputName.split(sep="_")[0]
+    config = inputName.split(sep="_")[1]
+    trial = inputName.split(sep="_")[3].split(sep='.')[0]
```

Comment:
"trial" is not used in the data class. please either remove or add to the data class